src/services/doppler-api-client.ts

```diff
@@ -180,7 +240,12 @@ export class HttpDopplerApiClient implements DopplerApiClient {
         unsubscriptionType: response.data.unsubscriptionType,
         manualUnsubscriptionReason: response.data.manualUnsubscriptionReason,
         unsubscriptionComment: response.data.unsubscriptionComment,
-        status: response.data.status,
+        status: this.getSubscriberStatus(
```

Comment:
Here you could call the two functions for suscriber status and unsuscription type to avoid having a nested funcion in `getSubscriberStatus` like this:
```
status:
          this.getSubscriberStatus(
            response.data.status,
            response.data.unsubscriptionType,
            response.data.manualUnsubscriptionReason,
            response.data.belongsToLists,
          ) +
            response.data.status !==
          'unsubscribed'
            ? this.getunsubscribedStatus(
                response.data.unsubscriptionType,
                response.data.manualUnsubscriptionReason,
              )
            : '',
```
And then remove unsuscribed call for function in `getSubscriberStatus`. What do you think?

Reply:
I'm not sure if this is best or not. Just a suggestion.

Reply:
Other option
```
status: this.getSubscriberStatus(
          response.data.status,
          response.data.unsubscriptionType,
          response.data.status === 'unsubscribed'
            ? this.getunsubscribedStatus(
                response.data.unsubscriptionType,
                response.data.manualUnsubscriptionReason,
              )
            : '',
        ),
``` 
Here you remove the `getunsubscribedStatus` function from inside `getSubscriberStatus` and also can remove two parameters  `response.data.unsubscriptionType` and `response.data.manualUnsubscriptionReason`
But add an extra parameter in `getSubscriberStatus` --> unsuscriberStatus

This is very fun and all but I'm done with this 😄 Thanks for your patience.